### PR TITLE
HCAP-531 prettier config

### PR DESCRIPTION
**Pull request by arranfw**
https://freshworks.atlassian.net/browse/HCAP-531

Throwing this up as a first step to get a consensus on the configuration we want. 

Let's hear your thoughts!

### CODE REVIEW

**Comment by ashtonmeuser** [comment]
Assuming this is incomplete as of yet and Prettier will eventually be a dev dep. This may also have to follow in the way of our linting and have separate config for server and client.

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "arrowParens": "avoid",
```
Comment (ashtonmeuser) [inline .prettierrc:2]: This will clash with AirBNB lint config.

Comment (ashtonmeuser) [reply .prettierrc:2]: I think. Assuming this is referring to arrow func param parens.

**Review by dbayly-freshworks — APPROVED** [review]

File: .prettierrc (hunk 1 above)
Comment (arranfw) [reply .prettierrc:2]: Good shout

**Comment by arranfw** [comment]
> Assuming this is incomplete as of yet and Prettier will eventually be a dev dep. This may also have to follow in the way of our linting and have separate config for server and client.

Yeah this is just meant to discuss the config, rest is incoming. Do we intentionally want different rules for FE and BE?

**Review by Zenen-freshworks — DISMISSED** [review]
I looked through, I don't know much about prettier but everything seems properly put together!

File: package.json [hunk 2]
```diff
@@ -1,8 +1,20 @@
...
+    "format:lint": "run-s 'lint:*'",
+    "lint:client": "cd client && npm run lint",
+    "lint:server": "cd server && npm run lint"
+  },
   "devDependencies": {
     "cypress": "6.1.0",
     "cypress-dotenv": "1.2.2",
-    "dotenv": "8.2.0"
+    "dotenv": "8.2.0",
+    "eslint-config-prettier": "^8.3.0",
```
Comment (fwkendall) [inline package.json:14]: can we lock all these deps please? 🙏

**Review by fwkendall — APPROVED** [review]

File: client/public/ie.js [hunk 3]
```diff
@@ -1,22 +1,25 @@
...
-window.onload = function() {
-  if (browser.name === "MSIE" || browser.name === "IE" ) {
-    var tag = document.createElement("p");
+window.onload = function () {
+  if (browser.name === 'MSIE' || browser.name === 'IE') {
+    var tag = document.createElement('p');
     tag.id = 'ie-browser';
-    tag.textContent="This form does not work on Internet Explorer. Please consider using Google Chrome or Firefox instead.";
+    tag.textContent =
+      'This form does not work on Internet Explorer. Please consider using Google Chrome or Firefox instead.';
```
Comment (fwkendall) [inline client/public/ie.js:17]: We should get Rachit to re-test this feature in IE FYI - we've tried to apply "nice" modern formatting here in the past and discovered it didn't work, although it wasn't specifically single quotes so we might be fine 😄

**Comment by Zenen-freshworks** [comment]
oooee, this one's a doozy! I'm going to have faith in the prettier process and say that it didn't break any niche features, but I suppose time will tell.

**Review by Zenen-freshworks — APPROVED** [review]

**Comment by arranfw** [comment]
> oooee, this one's a doozy! I'm going to have faith in the prettier process and say that it didn't break any niche features, but I suppose time will tell.

That's basically the exact mantra for adding prettier to an established project 😄

**Comment by arranfw** [comment]
TIL, don't try to configure CRA eslint, it's all implemented magically in the background

**Review by fwkendall — APPROVED** [review]

**Review by ashtonmeuser — APPROVED** [review]

**Pull request merged**